File: implement-rand10-using-rand7/implement-rand10-using-rand7.cpp
// The rand7() API is already defined for you.
// int rand7();
// @return a random integer in the range 1 to 7

class Solution {
public:
    int rand10() {
        int index=INT_MAX;
        while (index>=40){
            index=rand7()-1+(rand7()-1)*7;
        }
        return 1+(index%10);
    }
};

File: implement-rand10-using-rand7/implement-rand10-using-rand7.py
# The rand7() API is already defined for you.
# def rand7():
# @return a random integer in the range 1 to 7

class Solution:
    def rand10(self):
        """
        :rtype: int
        """
        c = (rand7() - 1)*7 + rand7() - 1
        return self.rand10() if c >= 40 else (c % 10) + 1
